### ISSUE TITLE
Not compatible with ActiveJob: Error: Message must include a Sidekiq::Worker class

### ISSUE THREAD

**Issue by andrewhavens** [moove-it/sidekiq-scheduler#79]
When I click the new "enqueue now" button in the UI, I get an error:

> ArgumentError at /sidekiq/recurring-jobs/test_notification/enqueue
> Message must include a Sidekiq::Worker class, not class name: [PushNotifications::SendNotificationJob, PushNotifications::BaseJob, ActiveJob::Base, ActiveJob::Translation, ActiveJob::Logging, ActiveJob::Callbacks, ActiveSupport::Callbacks, ActiveJob::Execution, ActiveSupport::Rescuable, ActiveJob::Enqueuing, ActiveJob::QueueName, ActiveJob::QueueAdapter, ActiveJob::Core, Object, PP::ObjectMixin, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, Kernel, BasicObject]

The whole reason I'm trying to manually enqueue the job is because I can't get any of my recurring jobs to run at all.

Here is my configuration yaml:
```yaml
test_notification:
  every: 3m
  class: PushNotifications::SendNotificationJob
  args: ["2c56d016-ddbd-4537-8a19-4bf4d2a2e462", "Test Notification!"]
```

I'm using the latest version of Sidekiq (4.0.2) and the latest commit on the master branch of this repo. I'm trying to test this in my development environment. So I started sidekiq with `bundle exec sidekiq`. The output shows:

```
2016-01-07T21:04:14.777Z 23500 TID-ox04q3th4 INFO: Running in ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
2016-01-07T21:04:14.777Z 23500 TID-ox04q3th4 INFO: See LICENSE and the LGPL-3.0 for licensing details.
2016-01-07T21:04:14.778Z 23500 TID-ox04q3th4 INFO: Upgrade to Sidekiq Pro for more features and support: http://sidekiq.org
2016-01-07T21:04:14.778Z 23500 TID-ox04q3th4 INFO: Booting Sidekiq 4.0.2 with redis options {:url=>nil}
2016-01-07T21:04:14.780Z 23500 TID-ox04q3th4 INFO: Loading Schedule
2016-01-07T21:04:14.780Z 23500 TID-ox04q3th4 INFO: Schedule empty! Set Sidekiq.schedule
2016-01-07T21:04:14.781Z 23500 TID-ox04q3th4 INFO: Schedules Loaded
2016-01-07T21:04:14.787Z 23500 TID-ox04q3th4 INFO: Starting processing, hit Ctrl-C to stop
```

Am I doing something wrong?

**Comment by elpic** [comment]
Hi @andrewhavens, 

How are you doing?

Is the full yaml? Or is a part of it?

Please show me how do you load that yaml.

Thanks

**Comment by andrewhavens** [comment]
My example YAML is only one of my jobs. I can try removing the others to narrow it down.

My initializer looks like this:
```ruby
require 'sidekiq/scheduler'

Sidekiq.configure_server do |config|
  config.on(:startup) do
    Sidekiq.schedule = YAML.load_file(File.expand_path("../../../config/sidekiq_schedule.yml", __FILE__))
  end
end
```

It seems to be loading the schedule because it appears in the Sidekiq UI.

**Comment by andrewhavens** [comment]
I wonder, it is raising an error because my class extends from `ActiveJob::Base`?

**Comment by elpic** [comment]
Could you load it from sidekiq.yml instead. I think works better that way.

**Comment by elpic** [comment]
You can add a key called schedule to your sidekiq.yml like this and that will be loaded automatically

```
:schedule:
  job:
    every: 4h
    class: MyWorker
```

**Comment by andrewhavens** [comment]
Ah, yes, adding that to `sidekiq.yml` seemed to get it to work. I did not previously have a `sidekiq.yml` file, but I created one with the following contents:

```yml
:schedule:
  job:
    every: 30s
    class: PushNotifications::SendNotificationJob
    args: ["2c56d016-ddbd-4537-8a19-4bf4d2a2e462", "Test Notification!"]
```

Now the jobs are automatically running, but they are still failing for the same reason. Here is the output from the sidekiq log:

```
2016-01-07T23:54:55.337Z 25918 TID-oweq2fkzc INFO: Running in ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
2016-01-07T23:54:55.337Z 25918 TID-oweq2fkzc INFO: See LICENSE and the LGPL-3.0 for licensing details.
2016-01-07T23:54:55.337Z 25918 TID-oweq2fkzc INFO: Upgrade to Sidekiq Pro for more features and support: http://sidekiq.org
2016-01-07T23:54:55.338Z 25918 TID-oweq2fkzc INFO: Booting Sidekiq 4.0.2 with redis options {:url=>nil}
2016-01-07T23:54:55.350Z 25918 TID-oweq2fkzc INFO: Loading Schedule
2016-01-07T23:54:55.350Z 25918 TID-oweq2fkzc INFO: Scheduling job 
2016-01-07T23:54:55.351Z 25918 TID-oweq2fkzc INFO: Schedules Loaded
2016-01-07T23:54:55.351Z 25918 TID-oweq2fkzc INFO: Starting processing, hit Ctrl-C to stop
2016-01-07T23:55:25.614Z 25918 TID-owerpsjqk INFO: queueing PushNotifications::SendNotificationJob (job)
2016-01-07T23:55:25.615Z 25918 TID-owerpsjqk INFO: ArgumentError: Message must include a Sidekiq::Worker class, not class name: [PushNotifications::SendNotificationJob, PushNotifications::BaseJob, ActiveJob::Base, ActiveJob::Translation, ActiveJob::Logging, ActiveJob::Callbacks, ActiveSupport::Callbacks, ActiveJob::Execution, ActiveSupport::Rescuable, ActiveJob::Enqueuing, ActiveJob::QueueName, ActiveJob::QueueAdapter, ActiveJob::Core, Object, PP::ObjectMixin, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, Kernel, BasicObject]
```

**Comment by andrewhavens** [comment]
Looks like this error comes from this line in Sidekiq: https://github.com/mperham/sidekiq/blob/6ad6a3aa330deebd76c6cf0d353f66abd3bef93b/lib/sidekiq/client.rb#L225

**Comment by andrewhavens** [comment]
I'm guessing it has something to do with how sidekiq-scheduler is trying to queue my ActiveJob jobs directly with Sidekiq instead of using the ActiveJob adapter: https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb

**Comment by elpic** [comment]
Yes, you are right I will be working on a fix for this. Thank you.

**Comment by andrewhavens** [comment]
I was able to get it to run with this configuration. Not ideal, but it's a start.

Note that I did not have to pass all of the options to the wrapper job. I assume that it used some defaults. I'm not sure if `wrapped` is necessary either.

```yaml
:schedule:
  unread_count_job:
    every: 30s
    class: ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper
    wrapped: PushNotifications::SendNotificationJob
    args:
      -
        job_class: PushNotifications::SendNotificationJob
        # job_id:     job_id,
        # queue_name: queue_name,
        # priority:   priority,
        arguments: ["533fa9ab-57e2-4c96-9b36-3f9c45ba317a", "Test Notification!"]
        # locale:     I18n.locale
```

**Comment by elpic** [comment]
Fix on released v2.0.2.
